samza-sql/src/main/java/org/apache/samza/sql/translator/ScanTranslator.java

```diff
@@ -169,15 +174,26 @@ void translate(final TableScan tableScan, final String queryLogicalId, final Str
       return;
     }
 
-    KVSerde<Object, Object> noOpKVSerde = KVSerde.of(new NoOpSerde<>(), new NoOpSerde<>());
-    DelegatingSystemDescriptor
-        sd = systemDescriptors.computeIfAbsent(systemName, DelegatingSystemDescriptor::new);
-    GenericInputDescriptor<KV<Object, Object>> isd = sd.getInputDescriptor(streamId, noOpKVSerde);
+    // set the wrapper input transformer (SamzaSqlInputTransformer) in system descriptor
+    DelegatingSystemDescriptor sd = systemDescriptors.get(systemName);
+    if (sd == null) {
+      sd = new DelegatingSystemDescriptor(systemName, new SamzaSqlInputTransformer());
+      systemDescriptors.put(systemName, sd);
+    } else {
+      /* in SamzaSQL, there should be no systemDescriptor setup by user, so this branch happens only
```

Comment:
Now transformer can be set in construction of the DelegatingSystemDescriptor, I throw error if Transformer is set to something other than SamzaSqlInputTransformer. And this path is hit in case of multiple usage of an input stream (as in the comment).